GAD 8 - Binomial Heap/tests/binomilia/BinomialHeapTester.java

```diff
@@ -0,0 +1,2282 @@
+
+        assertEquals(-799, heap.min(), "You returned wrong value of min! Expected <-799> but got <" + heap.min() + ">");
+        if (expectedLogs.size() < fastResult.getLogs().size()) {
+            fail("You logged too much! Expected logs (size: " + expectedLogs.size() + ") <\n" + expectedLogs + "\n> but got (size: " + fastResult.getLogs() + ") <\n" + fastResult.getLogs().toString() + "\n>");
+        } else if (expectedLogs.size() > fastResult.getLogs().size()) {
+            fail("You didn't log enough! Expected logs (size: " + expectedLogs.size() + ") <\n" + expectedLogs + "\n> but got (size: " + fastResult.getLogs() + ") <\n" + fastResult.getLogs().toString() + "\n>");
+        }
+        assertEquals(expectedLogs, fastResult.getLogs());
+    }
+
+    /**
+     * <p>
+     * Simple tests to see if the code throws an exception when possible errors could occur.<br>
+     * </p>
+     * Tests {@link BinomialHeap#deleteMin(Result)} and {@link BinomialHeap#min()} <br>
+     * <i> Note: Read the class docs </i>
+     * @author Aamin
+     */
+    @Test
+    public void exceptions() {
+        BinomialHeap binomilia = new BinomialHeap();
+
+        assertThrows(NoSuchElementException.class, binomilia::min);
+
+        binomilia.insert(69, new FastResult());
+        binomilia.deleteMin(new FastResult());
+        assertThrows(NoSuchElementException.class, () -> binomilia.deleteMin(new FastResult()));
+        assertThrows(NoSuchElementException.class, binomilia::min);
+    }
+
+    /**
+     * <p>
+     * Simple tests to see if numbers can be remove after being added in ascending order in the range [1, 20) or not. <br>
+     * Only positive integers are added! <br>
+     * Deletes the minimum element until can't no more
+     * </p>
+     * This test works only when {@link BinomialHeap#insert(int, Result)} has been implemented and then tests based on that
+     * both {@link BinomialHeap#deleteMin(Result)} and {@link BinomialHeap#min()} <br>
+     * <i> Note: Read the class docs </i>
+     * @author Aamin
+     */
+    @Test
+    public void simpleDelete() {
+        BinomialHeap heap = new BinomialHeap();
+        FastResult fastResult;
+        List<String> expectedLogs;
+        int deletedMin;
+
+        heap.insert(1, new FastResult());
```

Comment:
Loops exist :) also see line #1251

Reply:
Ah true, I could've done that as well, my brain was too much into printing everything mode at that time, I'll convert it to a loop rn